subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultArtifactCaches.java

```diff
@@ -56,18 +58,22 @@ public DefaultArtifactCaches(CacheScopeMapping cacheScopeMapping,
             IncubationLogger.incubatingFeatureUsed("Shared read-only dependency cache");
             File baseDir = validateReadOnlyCache(documentationRegistry, new File(roCache));
             if (baseDir != null) {
+                readOnlyCacheBaseDir = baseDir;
                 readOnlyCacheMetadata = new DefaultArtifactCacheMetadata(cacheScopeMapping, baseDir);
```

Comment:
This code here is pretty bad, using `null`s to tie things together. I wonder if we could somehow not make it worse, though. It seems that `DefaultArtifactCacheMetadata` should be the one implementing the `GlobalCache` interface, no?